vmaas/reposcan/database/csaf_store.py

```diff
@@ -46,7 +47,132 @@ def _save_csaf_files(self, csaf_files: CsafFiles) -> list[int]:
         self.conn.commit()
         return db_ids
 
+    def _get_product_rows(self, products_by_status):
+        to_associate_ids = []
+        to_insert = defaultdict(list)
+        try:
+            cur = self.conn.cursor()
+            for status, products in products_by_status.items():
+                if not products:
+                    continue
+                rows = execute_values(cur, """SELECT id, cpe, package, module
+                                FROM csaf_products
+                                WHERE (cpe, package, module) in %s""", (products,), fetch=True)
+                for row in rows:
+                    to_associate_ids.append((row[0], status))
+                for product in products:
+                    insert = True
+                    for row in rows:
+                        if product == (row[1], row[2], row[3]):
+                            insert = False
+                            break
+                    if insert:
+                        to_insert[status].append(product)
+
+        except Exception as exc:
+            CSAF_FAILED_IMPORT.inc()
+            self.logger.exception("Failure while fetching CSAF products from the DB: %s", exc)
+        finally:
+            cur.close()
+
+        return to_insert, to_associate_ids
+
+    def _insert_missing_products(self, products_by_status: dict):
+        inserted_products = []
+        try:
+            cur = self.conn.cursor()
+            for status, products in products_by_status.items():
+                if products:
+                    ids = execute_values(cur, """
+                        INSERT INTO csaf_products (cpe, package, module)
+                        VALUES %s returning id;
+                        """, (products), fetch=True)
+                    inserted_products = [(result[0], status) for result in ids]
```

Comment:
only products from last iteration are added to `inserted_products`. I suppose it should be `inserted_products.extend([(result[0], status) for result in ids])`